homepage/homepage/app/(docs)/docs/[framework]/[[...slug]]/project-setup/providers/react-native-expo.mdx

```diff
@@ -0,0 +1,143 @@
+export const metadata = { title: "React Native Expo Providers" };
+
+import { CodeGroup } from "@/components/forMdx";
+
+# React Native Expo Providers
+
+## Introduction
+
+`<JazzProvider />` is the core component that connects your Expo application to Jazz. It handles:
+
+- **Data Synchronization**: Manages connections to peers and the Jazz cloud
+- **Local Storage**: Persists data locally between app sessions
+- **Schema Management**: Associates your data models with Jazz
+- **Authentication**: Connects your authentication system to Jazz
+
+## Setting up the provider
+
+Wrap your app components with the `<JazzProvider />` component:
+
+<CodeGroup>
+```tsx
+// App.tsx
+import { JazzProvider } from "jazz-expo";
+import { MyAppAccount } from "./schema";
+
+export function MyJazzProvider({ children }: { children: React.ReactNode }) {
+  return (
+    <JazzProvider
+      sync={{ peer: "wss://cloud.jazz.tools/?key=you@example.com" }}
+      AccountSchema={MyAppAccount}
+    >
+      {children}
+    </JazzProvider>
+  );
+}
+
+// Register the Account schema so `useAccount` returns our custom `MyAppAccount`
+declare module "jazz-expo" {
+  interface Register {
+    Account: MyAppAccount;
+  }
+}
+```
+</CodeGroup>
+
+## Provider Options
+
+- `kvStore`
+  - `ExpoSecureStoreAdapter` (default)
+- `AccountSchema`
+  - `Account` (default)
+- `CryptoProvider`
+  - `PureJSCrypto` (default) - Pure JavaScript crypto provider
+  - `RNQuickCrypto` - C++ accelerated crypto provider
+
+## Authentication in the Provider
+
+`<JazzProvider />` works with various authentication methods, with Passkey being the easiest way to get started. Passkeys are secure, simple for users, and work across devices.
+
+For detailed implementation, refer to our [Authentication Overview](/docs/authentication/overview) guide.
+
+<CodeGroup>
+```tsx
```

Comment:
Could you enable twoslash in these examples?

Having the typecheck would help us on catching this kind of errors in the examples before the review.

Reply:
Done.

Reply:
Actually, that hasn't helped. There's not quite enough context to avoid suppressing the errors